internal/services/maintenance/maintenance_assignment_dynamic_scope_resource.go

```diff
@@ -0,0 +1,417 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type MaintenanceDynamicScopeResource struct{}
+
+type Tag struct {
+	Tag    string   `tfschema:"tag"`
+	Values []string `tfschema:"values"`
+}
+
+type Filter struct {
+	Locations      []string `tfschema:"locations"`
+	OsTypes        []string `tfschema:"os_types"`
+	ResourceGroups []string `tfschema:"resource_groups"`
+	ResourceTypes  []string `tfschema:"resource_types"`
+	Tags           []Tag    `tfschema:"tags"`
+	TagFilter      string   `tfschema:"tag_filter"`
+}
+
+type MaintenanceDynamicScopeModel struct {
+	Name                       string   `tfschema:"name"`
+	MaintenanceConfigurationId string   `tfschema:"maintenance_configuration_id"`
+	Scope                      string   `tfschema:"subscription_id"`
+	Filter                     []Filter `tfschema:"filter"`
+}
+
+var _ sdk.Resource = MaintenanceDynamicScopeResource{}
+
+func (MaintenanceDynamicScopeResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"maintenance_configuration_id": {
+			Type:             pluginsdk.TypeString,
+			Required:         true,
+			ForceNew:         true,
+			ValidateFunc:     maintenanceconfigurations.ValidateMaintenanceConfigurationID,
+			DiffSuppressFunc: suppress.CaseDifference,
+		},
+
+		"subscription_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubscriptionID,
+		},
+
+		"filter": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"locations": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"os_types": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"resource_groups": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"resource_types": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Schema{
+							Type: pluginsdk.TypeString,
+						},
+					},
+					"tags": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"tag": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+								},
+								"values": {
+									Type:     pluginsdk.TypeList,
+									Required: true,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
This can be empty as the service uses it to mean that the given tag is unset, so we should remove this validation here.
```suggestion
```